app/controllers/application_controller.rb

```diff
@@ -7,7 +7,7 @@ class ApplicationController < ActionController::Base
 
     def check_token
       hashtext = JSON.parse(params.first[0])
-      user = User.find_by(google_credentials: hashtext['google_credentials'])
+      user = User.find_or_create_by(google_credentials: hashtext['google_credentials'])
```

Comment:
I think there shouldn't be a create here. This method is not supposed to create new user instances, it's only supposed to serve as a protection to avoid posting without the correct credential + token.